packages/react-hooks/src/useMergedRefs.ts

```diff
@@ -1,19 +1,24 @@
-import * as React from 'react';
+import { useRef, useCallback, Ref, MutableRefObject } from 'react';
 
 /**
  * React hook to merge multiple React refs (either MutableRefObjects or ref callbacks) into a single ref callback that
  * updates all provided refs
  * @param refs- Refs to collectively update with one ref value.
  */
-export function useMergedRefs<T>(...refs: React.Ref<T>[]): (instance: T) => void {
-  return React.useCallback((value: T) => {
-    refs.forEach(ref => {
+export function useMergedRefs<T>(...refs: Ref<T>[]): (instance: T) => void {
+  const state = useRef<(Ref<T> | undefined)[]>();
+
+  // Update refs list.
+  state.current = refs;
+
+  return useCallback((value: T) => {
```

Comment:
Coming in late here but this is exactly what `useConstCallback` was made for. It guarantees the callback never mutates. (The current implementation of `React.useCallback` also does that, but the React docs are clear that this may change in the future.)